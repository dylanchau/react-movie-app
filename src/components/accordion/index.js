import { createContext, useContext, useState } from 'react'

import {
  Body,
  Container,
  Frame,
  Header,
  Inner,
  Item,
  Title,
} from './styles/accordion'

const ToggleContext = createContext()

export default function Accordion({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = function AccordionTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>
}

Accordion.Item = function AccordionItem({ children, ...restProps }) {
  const [toggle, setToggle] = useState(false)

  return (
    <ToggleContext.Provider value={{ toggle, setToggle }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  )
}

Accordion.Header = function AccordionHeader({ children, ...restProps }) {
  const { toggle, setToggle } = useContext(ToggleContext)

  return (
    <Header onClick={() => setToggle((preState) => !preState)} {...restProps}>
      {children}
      {toggle ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  )
}

Accordion.Body = function AccordionBody({ children, ...restProps }) {
  const { toggle } = useContext(ToggleContext)
  return toggle ? <Body {...restProps}>{children}</Body> : null
}
