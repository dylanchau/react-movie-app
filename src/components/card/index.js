import { createContext, useContext, useState } from 'react'

import {
  Container,
  Content,
  Entities,
  Feature,
  FeatureClose,
  FeatureText,
  FeatureTitle,
  Group,
  Image,
  Item,
  Maturity,
  Meta,
  SubTitle,
  Text,
  Title,
} from './styles/card'

export const FeatureContext = createContext()

export default function Card({ children, ...restProps }) {
  const [showFeature, setShowFeature] = useState(false)
  const [itemFeature, setItemFeature] = useState({})

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...restProps}>{children}</Container>
    </FeatureContext.Provider>
  )
}

Card.Group = function CardGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Card.Tittle = function CardTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>
}

Card.SubTittle = function CardSubTittle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>
}

Card.Text = function CardText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Card.Meta = function CardMeta({ children, ...restProps }) {
  return <Meta {...restProps}>{children}</Meta>
}

Card.Image = function CardImage({ children, ...restProps }) {
  return <Image {...restProps} />
}

Card.Item = function CardItem({ item, children, ...restProps }) {
  const { setItemFeature, setShowFeature } = useContext(FeatureContext)

  return (
    <Item
      onClick={() => {
        setItemFeature(item)
        setShowFeature(true)
      }}
      {...restProps}
    >
      {children}
    </Item>
  )
}

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>
}

Card.Entities = function CardEntities({ children, ...restProps }) {
  return <Entities {...restProps}>{children}</Entities>
}

Card.Feature = function CardFeature({ children, category, ...restProps }) {
  const { showFeature, itemFeature, setShowFeature } = useContext(
    FeatureContext
  )
  return showFeature ? (
    <Feature
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
      {...restProps}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature((prev) => !prev)}>
          <Image src="/images/icons/close.png" alt="Close" />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alginItem="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {itemFeature.genre.charAt(0).toUpperCase() +
              itemFeature.genre.slice(1)}
          </FeatureText>
        </Group>
        {children}
      </Content>
    </Feature>
  ) : null
}
