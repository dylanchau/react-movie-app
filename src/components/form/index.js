import {
  Base,
  Container,
  Error,
  Input,
  Link,
  Submit,
  Text,
  TextSmall,
  Title,
} from './styles/form'

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Form.Title = ({ children, ...restProps }) => (
  <Title {...restProps}>{children}</Title>
)

Form.Base = ({ children, ...restProps }) => (
  <Base {...restProps}>{children}</Base>
)

Form.Text = ({ children, ...restProps }) => (
  <Text {...restProps}>{children}</Text>
)

Form.TextSmall = ({ children, ...restProps }) => (
  <TextSmall {...restProps}>{children}</TextSmall>
)

Form.Submit = ({ children, ...restProps }) => (
  <Submit {...restProps}>{children}</Submit>
)

Form.Error = ({ children, ...restProps }) => (
  <Error {...restProps}>{children}</Error>
)

Form.Link = ({ children, ...restProps }) => (
  <Link {...restProps}>{children}</Link>
)

Form.Input = ({ children, ...restProps }) => (
  <Input {...restProps}>{children}</Input>
)
