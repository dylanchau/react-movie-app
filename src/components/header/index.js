import { useState } from 'react'
import { Link as ReactRouterLink } from 'react-router-dom'

import {
  Background,
  ButtonLink,
  Container,
  Dropdown,
  Feature,
  FeatureCallOut,
  Group,
  Link,
  Logo,
  Picture,
  PlayButton,
  Profile,
  Search,
  SearchIcon,
  SearchInput,
  Text,
} from './styles/header'

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>
}

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>
}

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((preProp) => !preProp)}
        data-testid="search-click"
      >
        <img alt="search" src="/images/icons/search.png" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
        placeholder="Search..."
        data-testid="search-input"
      />
    </Search>
  )
}

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>
}

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  )
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>
}

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>
}

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>
}

Header.TextLink = function LinkText({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>
}

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>
}

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>
}
