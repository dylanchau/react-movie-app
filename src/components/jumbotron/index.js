import { Inner } from './styles/jumbotron';

export default function Jumbotron() {
  return (
    <Inner>
      <p>Hello</p>
    </Inner>
  );
}
