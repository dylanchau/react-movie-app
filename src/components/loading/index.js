import { LockBody, Picture, ReleaseBody, Spinner } from './styles/loading'

function Loading({ src, ...restProps }) {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} alt={src} />
    </Spinner>
  )
}

Loading.ReleaseBody = function LoadingReleaseBody({ ...restProps }) {
  return <ReleaseBody {...restProps} />
}

export default Loading
