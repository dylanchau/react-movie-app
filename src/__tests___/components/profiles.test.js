import { render } from '@testing-library/react'
import { Profiles } from 'components'
import logo from 'logo.svg'

describe('<Profiles />', () => {
  it('Render Profiles component with populate data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who&#39;s watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture src={logo} data-testid="profile-picture" />
            <Profiles.Name>Test</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    )

    expect(getByText("Who's watching?")).toBeTruthy()
    expect(getByText('Test')).toBeTruthy()
    expect(getByTestId('profile-picture')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render Profiles component with populate data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who&#39;s watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid="profile-misc-picture" />
            <Profiles.Name>Test</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    )

    expect(getByText("Who's watching?")).toBeTruthy()
    expect(getByText('Test')).toBeTruthy()
    expect(getByTestId('profile-misc-picture')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })
})
