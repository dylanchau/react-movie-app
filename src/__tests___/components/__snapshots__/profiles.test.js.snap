// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> Render Profiles component with populate data 1`] = `
<div
  class="profiles__Container-sc-1rsg3ab-0 bfTffP"
>
  <h1
    class="profiles__Title-sc-1rsg3ab-1 gjxRuz"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-1rsg3ab-2 eMHtSM"
  >
    <li
      class="profiles__Item-sc-1rsg3ab-5 dZiOMW"
    >
      <img
        class="profiles__Picture-sc-1rsg3ab-4 eWngcD"
        data-testid="profile-picture"
        src="/images/users/logo.svg.png"
      />
      <p
        class="profiles__Name-sc-1rsg3ab-3 isVleg"
      >
        Test
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> Render Profiles component with populate data 2`] = `
<div
  class="profiles__Container-sc-1rsg3ab-0 bfTffP"
>
  <h1
    class="profiles__Title-sc-1rsg3ab-1 gjxRuz"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-1rsg3ab-2 eMHtSM"
  >
    <li
      class="profiles__Item-sc-1rsg3ab-5 dZiOMW"
    >
      <img
        class="profiles__Picture-sc-1rsg3ab-4 eWngcD"
        data-testid="profile-misc-picture"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-sc-1rsg3ab-3 isVleg"
      >
        Test
      </p>
    </li>
  </ul>
</div>
`;
